pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -522,3 +524,33 @@ func (s *Service) getFilteredSecurityGroupID(securityGroup infrav1.AWSResourceRe
 
 	return *sgs.SecurityGroups[0].GroupId, nil
 }
+
+func getLaunchTemplateInstanceMarketOptionsRequest(spotMarketOptions *infrav1.SpotMarketOptions) *ec2.LaunchTemplateInstanceMarketOptionsRequest {
+	if spotMarketOptions == nil {
+		// Instance is not a Spot instance
+		return nil
+	}
+
+	// Set required values for Spot instances
+	spotOptions := &ec2.LaunchTemplateSpotMarketOptionsRequest{}
+
+	// The following two options ensure that:
+	// - If an instance is interrupted, it is terminated rather than hibernating or stopping
+	// - No replacement instance will be created if the instance is interrupted
+	// - If the spot request cannot immediately be fulfilled, it will not be created
+	// This behaviour should satisfy the 1:1 mapping of Machines to Instances as
+	// assumed by the Cluster API.
+	spotOptions.SetInstanceInterruptionBehavior(ec2.InstanceInterruptionBehaviorTerminate)
+	spotOptions.SetSpotInstanceType(ec2.SpotInstanceTypeOneTime)
```

Comment:
Persistent option is not applicable for ec2 auto scaling, you can see the message from AWS console when you manually try to set the spot instance type, there is only one option.
`Persistent: not applicable for EC2 auto scaling`
`Specify a persistent request so that interrupted Spot Instances are requested again. If you do not specify a request type, EC2 makes a one-time request by default. Persistent requests are only supported when interruption behavior is set to either hibernate or stop.`

So we don't need to set spot instance as default is one-time.

Reply:
Interruption behavior, again in console, it won't allow to set. The message there:
`For one-time requests, only terminate is valid. `

Default for SetInstanceInterruptionBehavior is terminate so no need to set here.

Reply:
Comments need to cover these.
I don't think 1:1 matching is related to these settings, for each instance there will be a new Machine object when MachinePool Machine proposal is implemented.

Reply:
In summary, not all the options are valid for ec2 auto scaling.

Reply:
Thanks @sedefsavas , will take care of this.